frontend/src/components/tracing/tracing.tsx

```diff
@@ -134,23 +140,62 @@ interface VegaHoverCellSignal {
 
 const TraceBlock: React.FC<{
   run: Run;
+  /**
+   * undefined means the user hasn't clicked on this run yet
+   */
+  isExpanded: boolean | undefined;
+  isMostRecentRun: boolean;
   chartPosition: ChartPosition;
   theme: ResolvedTheme;
-}> = ({ run, chartPosition, theme }) => {
-  const [collapsed, setCollapsed] = useState(false);
-  const [hoveredCellId, setHoveredCellId] = useState<CellId | null>();
-
-  // To send signals to Vega from React, we bind a hidden input element
-  const hiddenInputRef = useRef<HTMLInputElement>(null);
-  const dispatchHoverEvent = (cellId: CellId | null) => {
-    // dispatch input event to trigger vega's param to update
-    if (hiddenInputRef.current) {
-      hiddenInputRef.current.value = String(cellId);
-      hiddenInputRef.current.dispatchEvent(
-        new Event("input", { bubbles: true }),
-      );
-    }
+}> = ({ run, isMostRecentRun, chartPosition, isExpanded, theme }) => {
+  const setExpandedRuns = useSetAtom(expandedRunsAtom);
+  const onToggleExpanded = () => {
+    setExpandedRuns((prev) => {
+      const newMap = new Map(prev);
+      newMap.set(run.runId, !newMap.get(run.runId));
+      return newMap;
+    });
   };
+  // We prefer the user's last click, but if they haven't clicked on this run,
+  // we expand the most recent run by default, otherwise we collapse it.
+  isExpanded = isExpanded ?? isMostRecentRun;
```

Comment:
I thought for perf; we could leave them collapsed, except the most recent one. but if the user ever toggles it, we then prefer their toggle state